torch/jit/_recursive.py

```diff
@@ -316,6 +315,8 @@ def create_script_module(nn_module, stubs_fn, share_types=True):
         concrete_type = concrete_type_builder.build()
     return create_script_module_impl(nn_module, concrete_type, stubs_fn)
 
+cached_modules = {}
```

Comment:
This is the scripted module instance cache I was referring to. I think it solves the problem I was facing in this PR but I'm worried about unintended consequences in other situations. If it seems okay, I can clean it up and make it more like the `ConcreteTypeStore`.